woocommerce/woocommerce
# Preload Jetpack data even without Jetpack plugin installed

Now that the Jetpack Connection package is a dependency of WooCommerce, preload Jetpack data even when the Jetpack plugin is not present.

Fixes #41070

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





WooCommerce has a feature that will simulate REST requests to get data and populate data stores with it on first load. This makes stores populated right away after render, without requiring extra HTTP requests.

This preloading logic has a condition that will populate Jetpack data from the `/jetpack/v4/connection` endpoint if the Jetpack plugin is present. If it is absent, it will still populate it but with values such as `isActive = false`, indicating that the Jetpack connection is disabled. This can be false when the Jetpack plugin is missing but a Jetpack connection has been established using the Jetpack Connection package, now present in WC's composer.json dependencies.

Places providing the data from the backend:
* https://github.com/woocommerce/woocommerce/blob/6ecf096d1c654e917c7052543b0811acebadc4bc/plugins/woocommerce/src/Internal/Admin/Loader.php
* https://github.com/woocommerce/woocommerce/blob/6ecf096d1c654e917c7052543b0811acebadc4bc/plugins/woocommerce/src/Internal/Admin/Settings.php

Place hydrating the data:
* https://github.com/woocommerce/woocommerce/blob/6ecf096d1c654e917c7052543b0811acebadc4bc/packages/js/data/src/plugins/with-plugins-hydration.tsx

The `/jetpack/v4/connection` endpoint is registered by the Jetpack Connection package so we can use it even without the Jetpack plugin present on the site.

This conditional preloading, alongside the recent removal of the dependency on Jetpack from the WCS&T onboarding tasks, currently causes a bug in at least one of these onboarding flows (ExperimentalShippingBanner), where the Jetpack connection is reported as disabled when it is in fact established.

This can be resolved by running `invalidateResolutionForStoreSelector` in the JS but I thought this is a cleaner solution.

Closes #41070.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Build WC from `trunk`, install & activate it.
2. Skip onboarding but set your store location to "US > California".
3. Go to WC Settings > Shipping.
4. Note the "Recommended shipping solutions" banner being presented at the bottom of the page:
   <img width="957" alt="Screenshot 2023-10-30 at 13 03 07" src="https://github.com/woocommerce/woocommerce/assets/1759681/23f7260b-1bd7-4360-882a-feeb79a8d468">
5. Click "Get started" in that banner. Go through the flow.
6. Return to WC Settings > Shipping.
7. Note the banner still appears and you can repeatedly go through it despite the Jetpack connection being there (with no Jetpack plugin).
8. Now follow the same steps but with this PR. The banner won't show up after the initial connection.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Preload Jetpack-related data from the Jetpack Connection package

#### Comment 

</details>
